testing/scenario_app/tool/logcat_reader.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+// It's bad to import a file from `bin` into `tool`.
+// However this tool is not very important, so delete it if necessary.
+import '../bin/utils/adb_logcat_filtering.dart';
+
+/// A tiny tool to read `adb logcat` output and perform some analysis.
+///
+/// This tool is not meant to be a full-fledged logcat reader. It's just a
+/// simple tool that uses the [AdbLogLine] extension type to parse results of
+/// `adb logcat` and explain what log tag names are most common.
+void main(List<String> args) {
+  if (args case [final String path]) {
+    final List<AdbLogLine> parsed = io.File(path)
```

Comment:
nit: Ideally this tool would by streaming from stdout.  That way it wouldn't be limited by memory size and you could say stuff like `tail foo.log | logcat_reader` or `adb logcat | logcat_reader`.

Reply:
Yeah this is intended to read from our stored logcat.txt, not necessarily the live process. I edited the tool to make that more clear!